src/content/ai/genui/get-started.md

```diff
@@ -0,0 +1,463 @@
+`A2uiContentGenerator` (from the `genui_a2ui` package)
+as examples of how to do so.
+
+## Create the connection to an agent
+
+If you build your Flutter project for iOS or macOS,
+add this key to your `{ios,macos}/Runner/*.entitlements` file(s)
+to enable outbound network requests:
+
+```xml
+<dict>
+...
+<key>com.apple.security.network.client</key>
+<true/>
+</dict>
+```
+
+Next, use the following instructions to connect your app
+to your chosen agent provider.
+
+1. Create a `GenUiManager`, and provide it with the catalog
+   of widgets that you want to make available to the agent.
+
+2. Create a `ContentGenerator`, and provide it with a
+   system instruction and a set of tools (functions
+   you want the agent to be able to invoke).
+   You should always include those provided by `GenUiManager`,
+   but feel free to include others.
+
+3. Create a `GenUiConversation` using the instances of `ContentGenerator`
+   and `GenUiManager`. Your app will primarily interact with this object
+   to get things done.
+
+   For example:
+
+```dart
+class _MyHomePageState extends State<MyHomePage> {
+  late final GenUiManager _genUiManager;
+  late final GenUiConversation _genUiConversation;
+
+  @override
+  void initState() {
+    super.initState();
+
+    // Create a GenUiManager with a widget catalog.
+    // The CoreCatalogItems contain basic widgets for text, markdown, and images.
+    _genUiManager = GenUiManager(catalog: CoreCatalogItems.asCatalog());
+
+    // Create a ContentGenerator to communicate with the LLM.
+    // Provide system instructions and the tools from the GenUiManager.
+    final contentGenerator = FirebaseAiContentGenerator(
+      systemInstruction: '''
+        You are an expert in creating funny riddles. Every time I give you a word,
+        you should generate UI that displays one new riddle related to that word.
+        Each riddle should have both a question and an answer.
+        ''',
+      tools: _genUiManager.getTools(),
+    );
+
+    // Create the GenUiConversation to orchestrate everything.
+    _genUiConversation = GenUiConversation(
+      genUiManager: _genUiManager,
+      contentGenerator: contentGenerator,
+      onSurfaceAdded: _onSurfaceAdded, // Added in the next step.
+      onSurfaceDeleted: _onSurfaceDeleted, // Added in the next step.
+    );
+  }
+
+  @override
+  void dispose() {
+    _textController.dispose();
+    _genUiConversation.dispose();
+
+    super.dispose();
+  }
+}
+```
+   
+## Send messages and display the agent's responses
+
+Send a message to the agent using the `sendRequest` method
+in the `GenUiConversation` class.
+
+To receive and display generated UI:
+
+1. Use the callbacks in `GenUiConversation` to track the addition
+  and removal of UI surfaces as they are generated.
+  These events include a _surface ID_ for each surface.
+
+2. Build a `GenUiSurface` widget for each active surface using
+  the surface IDs received in the previous step.
+
+  For example:
+
+```dart
```

Comment:
meta question: how do we keep all of these code snippets up to date? The APIs will change over time.